plugins/scaffolder-backend-module-github/src/actions/githubPages.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputError } from '@backstage/errors';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from 'octokit';
+import {
+  createTemplateAction,
+  parseRepoUrl,
+} from '@backstage/plugin-scaffolder-node';
+import { examples } from './githubPages.examples';
+import { getOctokitOptions } from '@backstage/plugin-scaffolder-backend-module-github';
+
+/**
+ * Creates a new action that enables GitHub Pages for a repository.
+ *
+ * @public
+ */
+export function createGithubPagesAction(options: {
+  integrations: ScmIntegrationRegistry;
+  githubCredentialsProvider?: GithubCredentialsProvider;
+}) {
+  const { integrations, githubCredentialsProvider } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    buildType?: 'legacy' | 'workflow';
+    sourceBranch?: string;
+    sourcePath?: '/' | '/docs';
+    token?: string;
+  }>({
+    id: 'github:pages',
```

Comment:
should this be `github:pages:enable` or something instead?

Reply:
Sure, that crossed my mind when I was working on this. Should I update the various file names too? Feel like I should as there's some conventions for it.

Reply:
Yeah, thinking that `createGithubPagesEnableAction` etc makes more sense now if we're thinking of renaming.